app/src/main/kotlin/com/getstrm/pace/catalogs/Collibra.kt

```diff
@@ -1,73 +1,120 @@
+    inner class Database(override val catalog: CollibraCatalog, id: String, dbType: String, displayName: String) :
         DataCatalog.Database(catalog, id, dbType, displayName) {
-            
-        override suspend fun getSchemas(): List<DataCatalog.Schema> {
-            val assets = catalog.client.query(ListSchemaIdsQuery(id)).execute().data!!.assets!!.filterNotNull()
-                .flatMap { schema ->
-                    schema.schemas
-                }
-            return assets.map {
-                Schema(catalog, this, it.target.id.toString(), it.target.fullName)
+
+        override suspend fun listSchemas(pageParameters: PageParameters): PagedCollection<DataCatalog.Schema> {
+            val schemas = pagedCalls(pageParameters) { skip, pageSize ->
+                client.query(ListSchemaIdsQuery(id, skip, pageSize))
+                    .executeReturnValidData().assets.onlyNonNulls().first().schemas
             }
+            return schemas.map {
+                Schema(catalog, this, it.target.id.toString(), it.target.fullName)
+            }.withUnknownTotals()
+        }
+
+        override suspend fun getSchema(schemaId: String): DataCatalog.Schema {
+            val schemaAsset =
+                catalog.client.query(GetSchemaQuery(schemaId)).executeReturnValidData().assets.firstNonNull()
+            return Schema(catalog, this, schemaAsset.id.toString(), schemaAsset.fullName)
         }
     }
 
-    class Schema(private val catalog: CollibraCatalog, database: DataCatalog.Database, id: String, name: String) :
+    inner class Schema(private val catalog: CollibraCatalog, database: DataCatalog.Database, id: String, name: String) :
         DataCatalog.Schema(database, id, name) {
-        override suspend fun getTables(): List<DataCatalog.Table> =
-            catalog.client.query(ListTablesInSchemaQuery(id)).execute().data!!.assets!!.filterNotNull()
-                .flatMap { table ->
-                    table.tables.map { Table(catalog, this, it.target.id.toString(), it.target.fullName) }
-                }
+            
+        override suspend fun listTables(pageParameters: PageParameters): PagedCollection<DataCatalog.Table> {
+            val tables = pagedCalls(pageParameters) { skip, pageSize ->
+                // first() refers to the single schema
+                client.query(ListTablesInSchemaQuery(id, skip, pageSize))
+                    .executeReturnValidData().assets.onlyNonNulls().first().tables
+            }.map { Table(catalog, this, it.target.id.toString(), it.target.fullName) }
+            return tables.withUnknownTotals()
+        }
+
+        override suspend fun getTable(tableId: String): DataCatalog.Table {
+            val table = catalog.client.query(GetTableQuery(tableId)).executeReturnValidData().assets.onlyNonNulls().firstOrNull() ?:
+            throw BadRequestException(
+                INVALID_ARGUMENT, BadRequest.getDefaultInstance(),
```

Comment:
In order to provide rich error messages, this `BadRequest` shouldn't be the default instance.